Collections/Collections.tex

```diff
@@ -1135,6 +1134,71 @@ \section{Some hints for using collections}
 Another nasty problem arises if you use a mutable object, \ie an object that can change its hash value over time, as an element of a \ct{Set} or as a key to a \ct{Dictionary}.
 Don't do this unless you love debugging!
 
+\section{Sorting collections}
+\label{sec:sorting}
+Basically, there are two selectors for sorting a collection: \mthind{ArrayedCollection}{sort} and \mthind{Collection}{sorted}.
+So when should you use which of them?
+Actually, \ct{sort} is only defined on certain specializations of \ct{Collection}, such as \ct{ArrayedCollection} or \ct{OrderedCollection}.
+This message performs an in-place-sort on the collection, \ie after sending \ct{sort} to a collection instance, the order of its elements will have been updated.
+Conversely, \ct{sorted:} performs an out-of-place sort that will not modify the original collection but will answer a sorted copy of the receiver.
+While an in-place-sort might reach higher performance in some situations, an out-of-place sort is the preferable option in most situations because \begin{enumerate*}[label={\alph*)}]\item it helps you avoid unintended side-effects that eventually can make your code harder to debug and understand\footnote{%
+    Actually, it is dangerous to modify an inline array or string from your code in-place (and thus forbidden since \sq 5.3).%
+} and \item it does not restrict the collection you use to be an array or an ordered collection\end{enumerate*}.
+
+The following examples illustrates the difference between in-place and out-of-place sorting:
+
+\begin{code}{@TEST |array|}
+array := {3. 5. 1}.
+
+"Out-of-place sort"
+array sorted. --> #(1 3 5)
+array. --> #(3 5 1)
+
+"In-place sort"
+array sort.
+array. --> #(1 3 5)
+\end{code}
+
+Sort methods use the message \mthind{Magnitude}{<=} to establish sort order, so by default, they can sort magnitudes such as numbers, characters, times \etc (see \charef{basic} on \ct{Magnitude}), but also other objects such as strings or points.
+However, what can you do if you want to sort objects that do not implement \ct{<=}, or if you would like to customize sort order?
```

Comment:
"customize sort order" -> "customize the sort order"

Reply:
```suggestion
However, what can you do if you want to sort objects that do not implement \ct{<=}, or if you would like to customize the sort order?
```